src/packages/recompose/utils/createHocFromMiddleware.js

```diff
@@ -0,0 +1,96 @@
+import { Component } from 'react'
+import createEagerFactory from '../createEagerFactory'
+
+const MIDDLEWARE_INFO = typeof Symbol === 'function'
+  ? Symbol()
+  : '@@recompose/middlewareInfo'
+
+const createComponentFromMiddleware = (middlewares, BaseComponent) => {
+  const factory = createEagerFactory(BaseComponent)
+  class CreateComponentFromMiddleware extends Component {
+    constructor(initialProps) {
+      super(initialProps)
+      this.pendingXformProps = [initialProps]
+      this.xformProps = []
+
+      this.xforms = middlewares.reduce((xforms, middleware, i) => {
+        const createXform = middleware({
+          getProps: () => this.xformProps[i]
+        })
+        xforms[i] = createXform({
+          update: (props, cb) => {
+            this.xformProps[i] = this.pendingXformProps[i]
+            this.pendingXformProps[i + 1] = props
+            if (i === middlewares.length - 1) {
+              this.applyPropUpdate(props, cb)
+            } else {
+              this.xforms[i + 1].update(props, cb)
+            }
+          }
+        })
+        return xforms
+      }, [])
+    }
+
+    state = { childProps: null }
+    _isMounted = false
+    didReceivePropUpdate = false
+
+    destroy() {
+      this.xforms.forEach(xform => {
+        if (typeof xform.destroy === 'function') {
+          xform.destroy()
```

Comment:
Middleware returns an object (right now I'm calling it `xform` for "transform" but that's not necessarily final) with a (required) `update` function. It may also have a `destroy` function to clean up subscriptions, though this isn't being used yet by any of the HOCs I've implemented so far. `mapPropsStream` is an example of an HOC that would need it.